src/backend/catalog/pg_depend.c

```diff
@@ -787,3 +797,117 @@ get_index_constraint(Oid indexId)
 
 	return constraintId;
 }
+
+/*
+ * depLockAndCheckObject
+ *
+ * Lock the object that we are about to record a dependency on.
+ * After it's locked, verify that it hasn't been dropped while we
+ * weren't looking.  If the object has been dropped, this function
+ * does not return!
+ */
+static void
+depLockAndCheckObject(const ObjectAddress *referenced)
+{
+	enum SysCacheIdentifier cacheId;
+	const char *objName;
+
+	switch (getObjectClass(referenced))
+	{
+		case OCLASS_SCHEMA:
+			cacheId = NAMESPACEOID;
+			objName = "namespace";
```

Comment:
why not
```suggestion
			objType = "namespace";
			objName = get_namespace_name(referenced->objectId);
```
for a more understandable error message?

Reply:
I do not want to add not-100%-necessary code, like this addional request to sys cache. The problem itself seems pretty rear. I consider current error message info is enough to provide good hints for the user.

Reply:
```sql
create function schema.function(a1 schema1.type, a2 schema2.type, a3 schema3.type, a4 schema4.type, a5 schema5.type, a6 schema6.type, a7 schema7.type, a8 schema8.type) returns setof schema9.type as $$ select i from generate_series(1,5)i; $$ language sql;
ERROR:  namespace 33367 was concurrently dropped
```
which of 10 namespaces?

Reply:
> this addional request to sys cache

This can only be done before an error message is issued.

Reply:
Once again, I think that overloading with more code may be a blocker when later we will try to push this change to Postgres upstream. As for GPDB, performance of the DDL commands may be not a big deal, while for upstream Postgres that most likely will be an issue. 

> This can only be done before an error message is issued.

That can be deduced by knowing which objects are still present, for ex by smth like:
```
select * from pg_namespace where nspname in ('schema1', 'schema2', 'schema3', 'schema4', 'schema5', 'schema6', 'schema7', 'schema8', 'schema9');
```